zebra-consensus/src/transaction/tests.rs

```diff
@@ -160,3 +164,33 @@ fn v5_coinbase_transaction_with_enable_spends_flag_fails_validation() {
         Err(TransactionError::CoinbaseHasEnableSpendsOrchard)
     );
 }
+
+#[tokio::test]
+async fn v5_transaction_is_rejected_before_nu5_activation() {
+    let network = Network::Mainnet;
+    let blocks = zebra_test::vectors::MAINNET_BLOCKS.iter();
+    let canopy = NetworkUpgrade::Canopy;
+
+    let state_service = service_fn(|_| async { unreachable!("Service should not be called") });
+    let script_verifier = script::Verifier::new(state_service);
+    let verifier = Verifier::new(network, script_verifier);
+
+    let transaction = fake_v5_transactions_for_network(network, blocks)
+        .next()
+        .expect("At least one fake V5 transaction in the test vectors");
+
+    let result = verifier
+        .oneshot(Request::Block {
+            transaction: Arc::new(transaction),
+            known_utxos: Arc::new(HashMap::new()),
+            height: canopy
```

Comment:
Can you also write a success test, but panic if NU5 doesn't have an activation height?

If you mark the test as [`#[should_panic]`](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic), we'll remember to activate it when we add the NU5 activation height.

We'll need separate testnet and mainnet success tests, because the NU5 activation height for testnet is set a long time before mainnet.

Reply:
Great idea! :tada: